### PR TITLE
Prevent race conditions by adding a random suffix to snowflake stage

### PR DESCRIPTION
**Pull request by ganesh-krishnan**
## What
Add a random string suffix to snowflake stage names. Additionally, clean up the stage file after upload

## Why
Currently snowflake uploads are susceptible to race conditions because we use the same stage filename for uploads. If two processes run at the same time, they interfere with each other.

### CODE REVIEW

File: lore/io/connection.py [hunk 1]
```diff
@@ -188,15 +190,21 @@ def insert(self, table, dataframe, batch_size=10 ** 5):
                     tmp = tempfile.NamedTemporaryFile(delete=False, suffix='.csv.gz')
                     tmp.close()
                     batch.to_csv(tmp.name, index=False, header=False, sep='|', na_rep='\\N', quoting=csv.QUOTE_NONE, compression='gzip')
-                    self._connection.connection.cursor().execute('PUT file://%(path)s @~/staged;' % {'path': tmp.name})
+                    letters = string.ascii_letters
```
Comment (sarchak) [inline lore/io/connection.py:193]: This seems ok curious to know if there is any reason not to use uuid.

File: lore/io/connection.py [hunk 2]
```diff
@@ -188,15 +190,21 @@ def insert(self, table, dataframe, batch_size=10 ** 5):
...
+                    suffix = ''.join(random.choice(letters) for i in range(20))
```
Comment (montanalow) [inline lore/io/connection.py:194]: Why not use a real uuid like `uuid.uuid4()`?

Comment (ganesh-krishnan) [reply lore/io/connection.py:194]: ohh didn't know there was something like that available. let me modify it.

File: lore/io/connection.py (hunk 1 above)
Comment (ganesh-krishnan) [reply lore/io/connection.py]: Pushed a fix.

**Comment by ganesh-krishnan** [comment]
will merge PR that @drdrsh create to fix tests in a bit as well.

**Pull request merged**